WooCommerce/Classes/AppDelegate.swift

```diff
@@ -503,14 +503,14 @@ extension AppDelegate {
         ServiceLocator.stores.synchronizeEntities(onCompletion: nil)
     }
 
-    /// Deauthenticates the user upon application password generation failure.
+    /// De-authenticates the user upon application password generation failure or WPCOM token expiry.
     ///
-    private func listenToApplicationPasswordGenerationFailureNotification() {
-        guard ServiceLocator.stores.isAuthenticatedWithoutWPCom else {
-            return
+    private func listenToAuthenticationFailureNotifications() {
+        if ServiceLocator.stores.isAuthenticatedWithoutWPCom {
+            ServiceLocator.stores.listenToApplicationPasswordGenerationFailureNotification()
+        } else {
+            ServiceLocator.stores.listenToWPCOMInvalidWPCOMTokenNotification()
         }
```

Comment:
Nit: we should keep `ServiceLocator.stores` as a local variable and use it within the method instead of using the singleton repeatedly.

Reply:
Done in 6205a6bf527ddf81bab0dd1e4191e3b9b5dff0bd